### ISSUE TITLE
build args assigned to the same random value

### ISSUE THREAD

**Issue by mlazze** [docker/compose#8907]
**Description**

I'm not sure how or why, but i get docker-compose args all set to the same value, instead of each one being separate.

Example:
```
build:
            context: ./serve
            args:
                A: ${A}
                B: ${B}
                C: ${C}
                D: ${D}
                E: ${E}
                F: ${F}
```

On docker-compose 2.1 i get A,B,C,D,E,F all set to the same value (which one changes every build), on docker-compose 2.0.1 i get A=A, B=B, ...

I build a simple repo with docker-compose/Dockerfile and Makefile to facilitate testing.

What that does is:
1. set A,B,C,D,E,F to be respectively A,B,C,D,E,F via `export A=A` ...
2. in docker-compose we set build-args A,B,C,D,E,F to the previously set envvars
3. in Dockerfile we use ARG A, ARG B, ... to get the build vars
4. we echo the args in a file and read it later

**Steps to reproduce the issue:**
1. git clone https://gitlab.com/mlazze/test-docker
2. run `make run`

**Describe the results you received:**

You get `A:F|B:F|C:F|D:F|E:F|F:F` or `A:D|B:D|C:D|D:D|E:D|F:D`

**Describe the results you expected:**

You should get `A:A|B:B|C:C|D:D|E:E|F:F`

**Additional information you deem important (e.g. issue happens only occasionally):**

This only happens in docker-compose 2.1.0, not on docker-compose 2.0.1

**Output of `docker compose version`:**

```
Docker Compose version 2.1.0
```

**Output of `docker info`:**

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Build with BuildKit (Docker Inc., v0.6.1-docker)
  compose: Docker Compose (Docker Inc., 2.1.0)

Server:
 Containers: 147
  Running: 39
  Paused: 0
  Stopped: 108
 Images: 1342
 Server Version: 20.10.10
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: false
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 8686ededfc90076914c5238eb96c883ea093a8ba.m
 runc version: v1.0.2-0-g52b36a2d
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
  cgroupns
 Kernel Version: 5.14.16-arch1-1
 Operating System: Antergos Linux
 OSType: linux
 Architecture: x86_64
 CPUs: 12
 Total Memory: 15.56GiB
 Name: mlazze-pc
 ID: 2YRP:PHED:ENZ7:LVK5:YMDW:JGXE:S3AF:NJJY:FQHP:DLAI:6CEZ:L624
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Username: staskij
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

**Additional environment details:**

**Comment by thaJeztah** [comment]
This sounds like a duplicate of https://github.com/docker/compose/issues/8882, which should be fixed by https://github.com/docker/compose/pull/8889, and is part of v2.1.1

**Comment by mlazze** [comment]
Can confirm, this is solved in 2.1.1. Thank you!

**Comment by thaJeztah** [comment]
Great!
